src/passes/rewrite-namespace-this-references.ts

```diff
@@ -25,7 +25,15 @@ function rewriteSingleScopeThisReferences(
     blockStatement: estree.BlockStatement, namespaceReference: string) {
   astTypes.visit(blockStatement, {
     visitThisExpression(path: NodePath<estree.ThisExpression>) {
-      path.replace(jsc.identifier(namespaceReference));
+      const parent = path.parent;
+      if (parent && parent.node.type !== 'MemberExpression') {
+        // When a namespace object is itself referenced with `this` but isn't
+        // used to reference a member of the namespace, rewrite the `this` to
+        // `void 0`:
+        path.replace(jsc.unaryExpression('void', jsc.literal(0)));
```

Comment:
We've been using `undefined` elsewhere, but we should be consistent either way.

I have a mild preference for `undefined` as it's clearer, and there's no way to redefine the toplevel undefined variable in an ES5 compliant browser. The only time `undefined` wouldn't be undefined is if you declared a scoped variable named `undefined`, there's no way for one file or function to change the meaning of `undefined` for another.

Reply:
SGTM, updated.